### PR TITLE
test: 1:1 swap with different owner definition

### PR DESCRIPTION
**Pull request by Langstra**
This should yield the same results as the GH3614.test.ts I think. The only thing that is different here is the entity definition.

### CODE REVIEW

File: tests/issues/GH3614-owner-defined-different.test.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  name!: string;
+
+  @OneToOne(() => Project, p => p.owner, {
+    owner: true,
+    wrappedReference: true,
+    nullable: true,
+  })
+  project1?: IdentifiedReference<Project>;
+
+  @OneToOne(() => Project, p => p.owner, {
```
Comment (B4nan) [inline tests/issues/GH3614-owner-defined-different.test.ts:45]:
```suggestion
  @OneToOne(() => Project, p => p.secondaryOwner, {
```

**Comment by B4nan** [comment]
works fine if you fix that

**Comment by Langstra** [comment]
Hmm... seems I was sleeping. I'm trying to debug something in this area that it almost exactly similar to this code, but still gives me a `UNIQUE constraint failed`. However, I think I got the bug right now. It is related to the way I called `orm.em.create(User, {name: 'Johnny', project1: 1})`

**Comment by B4nan** [comment]
Maybe you have the same problem in your app, multiple owning properties mapped by a single inverse? I guess we could validate that.

**Comment by Langstra** [comment]
https://github.com/RiskChallenger/mikro-orm/commit/d356e96b3eb2582b08020051d16f05b35aa758a4 the only change compared to GH 3614 is on line 92 `const newOwner = orm.em.create(User, { name: 'Johnny', project1: 1 });` where I added `project1: 1`

**Comment by B4nan** [comment]
When you do this, you are setting the `project1` value on `newOwner`, which does not know anything about `oldOwner`.

```ts
newOwner.project1 = Reference.create(project);
```

When you do this, you literally replace old owner with new one, so it triggers the cleanup of the old owner.

```ts
project.owner = Reference.create(newOwner);
```

When you do this, the second line is a no-op, the relation is already set up, so the cleanup of `oldOwner` is not triggered.

```ts
const newOwner = orm.em.create(User, { name: 'Johnny', project1: project.id });
project.owner = Reference.create(newOwner);
```

**Comment by B4nan** [comment]
I don't feel like the fix I came up with is the correct solution, but it seems to be working and I am tired enough to just do it conditionally :D 

[389b4a2](https://github.com/mikro-orm/mikro-orm/commit/389b4a2a750a45ecb9806640613de016132d2dfa)

**Comment by Langstra** [comment]
Thanks for fixing. Would you like me to create an issue so it won't be forgotten and when you feel like it (or someone else) can fix it properly/differently?

**Comment by Langstra** [comment]
Hmm... sorry to bother you again on this one. Trying to switch the relation twice will fail. https://github.com/RiskChallenger/mikro-orm/blob/4fe88b130b384a49a81f158a53c06ece261b6c23/tests/issues/GH3614.test.ts#L106 Added a test case here on line 106.

**Comment by B4nan** [comment]
You are mixing contexts in there. You load the entity on a fresh fork, but call em.create on the global EM.

**Comment by Langstra** [comment]
I do not think I understand it. If I remove (comment) the entire part about the fresh fork the error still exists.

**Comment by B4nan** [comment]
You will need to wait, i wont be available till next week, so cant event test it now.

**Comment by Langstra** [comment]
Sorry to notify you, you get enough questions already and this one is in a closed PR. Would you like to have another look here or rather have me create a separate issue/PR with test case? Many thanks.

**Comment by B4nan** [comment]
Let's create an issue for that, it looks valid. Unlike my comment about the mixed contexts - the fork is used only to verify the removal of the entity, so that should be irrelevant.

**Comment by B4nan** [comment]
Ok, this should do it

**Pull request closed without merging**